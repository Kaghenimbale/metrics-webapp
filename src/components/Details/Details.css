.detail {
  display: flex;
  flex-direction: column;
  gap: 1rem;
}

.filteredItems {
  display: flex;
  flex-direction: column;
  gap: 1rem;
  padding: 0.6rem;
  background: rgb(19, 19, 19);
}

.daily-games h2 {
  margin: 0;
  text-align: center;
  color: white;
}

.arrow-left {
  font-size: 1.2rem;
  color: white;
}

.arrow-left-btn {
  background: none;
  border: none;
}

.image {
  width: 100%;
}

.icons {
  display: flex;
  gap: 0.8rem;
}

.image img {
  width: 100%;
}

.container {
  display: grid;
  grid-template-columns: 35vw auto;
  align-items: center;
  gap: 1rem;
}

.container h2,
.container p {
  margin: 0;
}

.container h2 {
  font-size: 1.2rem;
}

.description {
  color: darkgray;
}

.description a {
  color: lightgray;
}

.detail-nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 0.8rem;
  color: white;
}

.detail-left {
  display: flex;
  align-items: center;
}
