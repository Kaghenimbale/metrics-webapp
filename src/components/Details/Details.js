import React from 'react';
import { NavLink } from 'react-router-dom';
import { SlArrowLeft } from 'react-icons/sl';
import { useSelector } from 'react-redux';
import { MdKeyboardVoice } from 'react-icons/md';
import { FiSettings } from 'react-icons/fi';
import parse from 'html-react-parser';
import './Details.css';

const Details = () => {
  const { itemFilter } = useSelector((state) => state.scores);

  return (
    <div className="detail">
      <div className="detail-nav">
        <div className="detail-left">
          <NavLink to="/">
            <div className="arrow-container">
              <button className="arrow-left-btn" type="button">
                <SlArrowLeft className="icon-Item arrow-left" />
              </button>
            </div>
          </NavLink>
          <h3>2023</h3>
        </div>
        <h3>Today&#39;S Games</h3>
        <div className="icons">
          <MdKeyboardVoice className="icon-Item" />
          <FiSettings className="icon-Item" />
        </div>
      </div>
      <div className="daily-games">
        <h2>
          Today Game(
          {itemFilter.length}
          )
        </h2>
      </div>
      <div className="filteredItems">
        {
            itemFilter.map((item) => (
              <div className="content" key={item.videos[0].id}>
                <div className="container">
                  <div className="image">
                    { parse(item.videos[0].embed) }
                  </div>
                  <div className="description">
                    <h2><a href={item.competitionUrl}>{item.competition}</a></h2>
                    <h2>{item.title}</h2>
                    <span>{item.date}</span>
                    <p><a href={item.matchviewUrl}>{item.videos[0].title}</a></p>
                  </div>
                </div>
              </div>
            ))
        }
      </div>
    </div>
  );
};

export default Details;
