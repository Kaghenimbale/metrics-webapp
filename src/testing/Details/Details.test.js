import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import Details from '../../components/Details/Details';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureMockStore();

const store = mockStore({
  scores: {
    itemFilter: [
      {
        title: 'Liverpool - Nottingham Forest',
        competition: 'ENGLAND: Premier League',
        competitionUrl: 'https://www.example.com',
        date: '2023-04-23T12:34:56Z',
        matchviewUrl: 'https://www.example.com/matchview',
        videos: [
          { id: '64444f88d4c69', title: 'Highlights', embed: '<div></div>' },
        ],
      },
    ],
  },
});

describe('Details component', () => {
  it('renders the expected text', () => {
    render(
      <Provider store={store}>
        <Router>
          <Details />
        </Router>
      </Provider>,
    );

    expect(screen.getByText('Today Games(1)')).toBeInTheDocument();
    expect(screen.getByText('Liverpool - Nottingham Forest')).toBeInTheDocument();
    expect(screen.getByText('ENGLAND: Premier League')).toBeInTheDocument();
    expect(screen.getByText('2023')).toBeInTheDocument();
    expect(screen.getByText('Highlights')).toBeInTheDocument();
  });
  it('renders the score lists correctly', () => {
    const { container } = render(
      <Router>
        <Provider store={store}>
          <Details />
        </Provider>
      </Router>,
    );

    expect(container.querySelector('.game')).toMatchSnapshot();
  });
});
