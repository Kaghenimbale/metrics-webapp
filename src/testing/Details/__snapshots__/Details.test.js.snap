// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component renders the score lists correctly 1`] = `null`;
